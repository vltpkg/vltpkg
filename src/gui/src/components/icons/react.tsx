export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
