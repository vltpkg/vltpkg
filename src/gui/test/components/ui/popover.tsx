import { test, expect, afterEach } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from '@/components/ui/popover.jsx'

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  cleanup()
})

test('popover render default', async () => {
  const Container = () => (
    <Popover>
      <PopoverTrigger>Trigger Element</PopoverTrigger>
      <PopoverContent>Content</PopoverContent>
    </Popover>
  )
  render(<Container />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('popover render open', async () => {
  const Container = () => (
    <Popover open={true}>
      <PopoverTrigger>Trigger Element</PopoverTrigger>
      <PopoverContent>Content</PopoverContent>
    </Popover>
  )
  render(<Container />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})
