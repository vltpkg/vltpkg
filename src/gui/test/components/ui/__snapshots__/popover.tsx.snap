// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`popover render default 1`] = `

<div>
  <button
    type="button"
    aria-haspopup="dialog"
    aria-expanded="false"
    aria-controls="radix-:r0:"
    data-state="closed"
  >
    Trigger Element
  </button>
</div>

`;

exports[`popover render open 1`] = `

<span
  data-radix-focus-guard
  tabindex="0"
  style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
>
</span>
<div>
  <button
    type="button"
    aria-haspopup="dialog"
    aria-expanded="true"
    aria-controls="radix-:r1:"
    data-state="open"
  >
    Trigger Element
  </button>
</div>
<div
  data-radix-popper-content-wrapper
  style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content;"
>
  <div
    data-side="bottom"
    data-align="center"
    data-state="open"
    role="dialog"
    id="radix-:r1:"
    class="z-[10000] w-72 rounded-md border bg-popover p-4 text-popover-foreground shadow-md outline-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2"
    style="--radix-popover-content-transform-origin: var(--radix-popper-transform-origin); --radix-popover-content-available-width: var(--radix-popper-available-width); --radix-popover-content-available-height: var(--radix-popper-available-height); --radix-popover-trigger-width: var(--radix-popper-anchor-width); --radix-popover-trigger-height: var(--radix-popper-anchor-height); animation: none;"
    tabindex="-1"
  >
    Content
  </div>
</div>
<span
  data-radix-focus-guard
  tabindex="0"
  style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
>
</span>

`;
